ecosystem/typescript/sdk/src/aptos_client.ts

```diff
@@ -267,9 +269,31 @@ export class AptosClient {
   }
 
   /**
-   * Extracts event key from the account resource identified by the
-   * `event_handle_struct` and `field_name`, then returns events identified by the event key
-   * @param address Hex-encoded 32 byte Aptos account from which events are queried
+   * Event types are globally identifiable by an account `address` and
+   * monotonically increasing `creation_number`, one per event type emitted
+   * to the given account. This API returns events corresponding to that
+   * that event type.
+   * @param address Hex-encoded 32 byte Aptos account, with or without a `0x` prefix,
+   * for which events are queried. This refers to the account that events were emitted
+   * to, not the account hosting the move module that emits that event type.
+   * @param creationNumber Creation number corresponding to the event type.
+   * @returns Array of events assotiated with the given account and creation number.
+   */
+  @parseApiError
+  async getEventsByCreationNumber(
+    address: MaybeHexString,
+    creationNumber: number | BigInt | string,
```

Comment:
nit: BigInt is a wrapper type. Let's use `bigint` here. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt#type_information